File: Raspi/Freenove_RFID_Starter_Kit_for_Raspberry_Pi/Code/Python_GPIOZero_Code/06.2.1_Alertor/Alertor.py
#!/usr/bin/env python3
########################################################################
# Filename    : Alertor.py
# Description : Make Alertor with buzzer and button
# Author      : www.freenove.com
# modification: 2019/12/27
########################################################################
from gpiozero import TonalBuzzer,Button
from gpiozero.tones import Tone
import time
import math

buzzer = TonalBuzzer(17)
button = Button(18) # define Button pin according to BCM Numbering

def loop():
    while True:
        if button.is_pressed:  # if button is pressed
            alertor()
            print ('alertor turned on >>> ')
        else :
            stopAlertor()
            print ('alertor turned off <<<')
def alertor():
    buzzer.play(Tone(220.0)) 
    time.sleep(0.1)
        
def stopAlertor():
    buzzer.stop()
            
def destroy():
    buzzer.close()                  

if __name__ == '__main__':     # Program entrance
    print ('Program is starting...')
    try:
        loop()
    except KeyboardInterrupt:  # Press ctrl-c to end the program.
        destroy()
        print("Ending program")

File: Raspi/Freenove_RFID_Starter_Kit_for_Raspberry_Pi/Code/C_Code/06.2.1_Alertor/Alertor.c
/**********************************************************************
* Filename    : Alertor.c
* Description : Make Alertor with buzzer and button.
* Author      : www.freenove.com
* modification: 2019/12/27
**********************************************************************/
#include <wiringPi.h>
#include <stdio.h>
#include <softTone.h>
#include <math.h>

#define buzzerPin    0  	//define the buzzerPin
#define buttonPin 	 1		//define the buttonPin

void alertor(int pin){
	int x;
	double sinVal, toneVal;
	for(x=0;x<360;x++){ // frequency of the alertor is consistent with the sine wave 
		sinVal = sin(x * (M_PI / 180));		//Calculate the sine value
		toneVal = 2000 + sinVal * 500;		//Add the resonant frequency and weighted sine value 
		softToneWrite(pin,toneVal);			//output corresponding PWM
		delay(1);
	}
}
void stopAlertor(int pin){
	softToneWrite(pin,0);
}
int main(void)
{
	printf("Program is starting ... \n");
	
	wiringPiSetup();
	
	pinMode(buzzerPin, OUTPUT); 
	pinMode(buttonPin, INPUT);
	softToneCreate(buzzerPin); //set buzzerPin
	pullUpDnControl(buttonPin, PUD_UP);  //pull up to HIGH level
	while(1){	
		if(digitalRead(buttonPin) == LOW){ //button is pressed
			alertor(buzzerPin);   // turn on buzzer
			printf("alertor turned on >>> \n");
		}
		else {				//button is released 
			stopAlertor(buzzerPin);   // turn off buzzer
			printf("alertor turned off <<< \n");
		}
	}
	return 0;
}

